dao/src/main/java/org/thingsboard/server/dao/usage/BasicUsageInfoService.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright © 2016-2023 The Thingsboard Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.server.dao.usage;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+import org.thingsboard.server.common.data.ApiUsageRecordKey;
+import org.thingsboard.server.common.data.ApiUsageState;
+import org.thingsboard.server.common.data.UsageInfo;
+import org.thingsboard.server.common.data.id.TenantId;
+import org.thingsboard.server.common.data.kv.TsKvEntry;
+import org.thingsboard.server.common.data.tenant.profile.DefaultTenantProfileConfiguration;
+import org.thingsboard.server.dao.asset.AssetDao;
+import org.thingsboard.server.dao.customer.CustomerDao;
+import org.thingsboard.server.dao.dashboard.DashboardDao;
+import org.thingsboard.server.dao.device.DeviceDao;
+import org.thingsboard.server.dao.tenant.TbTenantProfileCache;
+import org.thingsboard.server.dao.timeseries.TimeseriesService;
+import org.thingsboard.server.dao.usagerecord.ApiUsageStateService;
+import org.thingsboard.server.dao.user.UserDao;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class BasicUsageInfoService implements UsageInfoService {
+
+    private final DeviceDao deviceDao;
+    private final AssetDao assetDao;
+    private final CustomerDao customerDao;
+    private final UserDao userDao;
+    private final DashboardDao dashboardDao;
+    private final ApiUsageStateService apiUsageStateService;
+    private final TimeseriesService tsService;
+    @Lazy
+    private final TbTenantProfileCache tenantProfileCache;
+
+    @Override
+    public UsageInfo getUsageInfo(TenantId tenantId) {
+        DefaultTenantProfileConfiguration profileConfiguration =
```

Comment:
This call may be quite time-consuming. Please consider caching some things that we can cache. Maybe we should cache the result of the "count" API calls unless the entity is added or removed. Maybe this should be done as a separate PR.

Reply:
Use TenantEntityCountsCache. Key: <TenantId, EntityType> Value: Integer